nmtwizard/prepoperator.py

```diff
@@ -0,0 +1,108 @@
+# coding: utf-8
+import six
+import abc
+import os
+
+from nmtwizard import tokenizer
+from nmtwizard import tu
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Operator(object):
+    """Base class for preprocessing opertators."""
+
+    @abc.abstractmethod
+    def __call__(self, tu_batch):
+        raise NotImplementedError()
+
+
+class PreprocessingPipeline(Operator):
+
+    def __init__(self):
+        self._ops = []
+
+    def add(self, op):
+        self._ops.append(op)
+
+    def __call__(self, tu_batch):
```

Comment:
I propose that `Operator.__call__` actually returns a `tu_batch` (could be the same list if the operation is in-place). This will make it easier to implement filtering operators.